### PR TITLE
feat(auditlog): add auto moderation actions and options

### PR DESCRIPTION
**Pull request by Earlopain**
https://discord.com/developers/docs/resources/audit-log#audit-log-entry-object-audit-log-events
https://discord.com/developers/docs/resources/audit-log#audit-log-entry-object-optional-audit-entry-info

### CODE REVIEW

File: structs.go [hunk 1]
```diff
@@ -1738,14 +1738,17 @@ const (
...
-	RoleName         string               `json:"role_name"`
+	DeleteMemberDays              string               `json:"delete_member_days"`
+	MembersRemoved                string               `json:"members_removed"`
+	ChannelID                     string               `json:"channel_id"`
+	MessageID                     string               `json:"message_id"`
+	Count                         string               `json:"count"`
+	ID                            string               `json:"id"`
+	Type                          *AuditLogOptionsType `json:"type"`
+	RoleName                      string               `json:"role_name"`
+	ApplicationID                 string               `json:"application_id"`
```
Comment (FedorLap2006) [inline structs.go:1749]:
```suggestion
```

Just to keep the commit history tidy, maybe a separate PR?

**Review by FedorLap2006 — CHANGES_REQUESTED** [review]
LGTM. However, please note the comment.

File: structs.go (hunk 1 above)
Comment (Earlopain) [reply structs.go]: I guess. I will open the other PR after this is merged

**Review by FedorLap2006 — APPROVED** [review]

**Comment by FedorLap2006** [comment]
Thanks for your contribution!

**Pull request merged**